### ISSUE TITLE
[bazel] Compare mask rom built by bazel with meson

### ISSUE THREAD

**Issue by alphan** [lowRISC/opentitan#12069]
Before we complete the transition from meson to bazel we should compare the mask roms built by bazel and meson. Some things that may be worth checking:
* All unit (off-target) and functional (on-target) tests should run and pass.
* Binary size.
* Symbols.
* Disassembly diff (spot checks).
* CLI arguments to toolchain (if this is relatively easy to obtain and compare in a meaningful way).

cc @arunthomas

**Comment by alphan** [comment]
[doc](https://docs.google.com/document/d/1p3Yqj4-iupAnJGPCzI_P3Pbc8X3hcCJl56D5MvbBJIo/edit)

**Comment by timothytrippel** [comment]
@alphan can we close this now? (I remember you mentioning you got things worked out such that both images are the same now?)

**Comment by alphan** [comment]
@timothytrippel yes, the only blocking issue (a minor difference in size, independent of optimization flags) was fixed in #12451.

The last time I checked, bazel was running all tests ran by meson (+ 4 additional tests) and I also did a handful of spot checks in disassembly.

Also, the last time I checked, functests were run by systemtest. I guess we can check those when we deprecate it. PLMK if we should create a separate issue for this.

I created the following issues for optimization flags, symbol visibility, annotated disassembly, and warnings:
* #12549
* #12551
* #12552
* #12553
